sklearn/utils/validation.py

```diff
@@ -32,44 +32,65 @@
 FLOAT_DTYPES = (np.float64, np.float32, np.float16)
 
 
-def _deprecate_positional_args(f):
+def _deprecate_positional_args(func=None, *, version="0.25"):
     """Decorator for methods that issues warnings for positional arguments.
 
     Using the keyword-only argument syntax in pep 3102, arguments after the
     * will issue a warning when passed as a positional argument.
 
     Parameters
     ----------
-    f : callable
-        Function to check arguments on.
+    version : callable or str, default="0.25"
+        The version when positional arguments will result in error. If
+        callable, then "0.25" will be used for error message.
```

Comment:
This is the docstring of the previous implementation, now it's only str and it's missing the func parameter.